query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/search_filter.rs

```diff
@@ -0,0 +1,123 @@
+use query_engine_tests::*;
```

Comment:
Missing tests for negation and non-negation + negation searches to validate that your filter merging doesn't do incorrect merges.